auth/jwt_test.go

```diff
@@ -107,279 +109,104 @@ func mockTokenWithSingleAudience() string {
-
-	jwt := jose.JWT{
-		RawHeader:  jws.RawHeader,
-		Header:     jws.Header,
-		RawPayload: jws.RawPayload,
-		Payload:    jws.Payload,
-		Signature:  jws.Signature}
-
-	// Check JWT issuer details
-	issuer, audiences, err := GetJWTIssuer(jwt)
-	assert.NoError(t, err)
-	assert.Nil(t, audiences)
-	assert.Equal(t, 0, len(audiences))
-	assert.Empty(t, issuer)
-}
-
-func TestGetJWTIssuerWithNoIssuer(t *testing.T) {
-	// Parse the mocked JWS token.
-	token := mockTokenWithNoIssuer()
-	jws, err := jose.ParseJWS(token)
-	assert.NotNil(t, jws)
-	assert.NoError(t, err)
-
-	// Verify the header, payload, and signature.
-	parts := strings.Split(token, ".")
-	assert.NotNil(t, parts)
-	assert.Equal(t, parts[0], jws.RawHeader)
-	assert.Equal(t, parts[1], jws.RawPayload)
-
-	assert.NotNil(t, jws.Header)
-	assert.NotNil(t, jws.Payload)
-	assert.NotNil(t, jws.Signature)
-
-	jwt := jose.JWT{
-		RawHeader:  jws.RawHeader,
-		Header:     jws.Header,
-		RawPayload: jws.RawPayload,
-		Payload:    jws.Payload,
-		Signature:  jws.Signature}
-
-	// Check JWT issuer details
-	issuer, audiences, err := GetJWTIssuer(jwt)
-	assert.Error(t, err)
-	assert.Nil(t, audiences)
-	assert.Equal(t, 0, len(audiences))
-	assert.Empty(t, issuer)
-}
-
-func TestGetJWTIssuerWithBadIssuer(t *testing.T) {
-	// Parse the mocked JWS token.
-	token := mockTokenWithBadIssuer()
-	jws, err := jose.ParseJWS(token)
-	assert.NotNil(t, jws)
-	assert.NoError(t, err)
-
-	// Verify the header, payload, and signature.
-	parts := strings.Split(token, ".")
-	assert.NotNil(t, parts)
-	assert.Equal(t, parts[0], jws.RawHeader)
-	assert.Equal(t, parts[1], jws.RawPayload)
-
-	assert.NotNil(t, jws.Header)
-	assert.NotNil(t, jws.Payload)
-	assert.NotNil(t, jws.Signature)
-
-	jwt := jose.JWT{
-		RawHeader:  jws.RawHeader,
-		Header:     jws.Header,
-		RawPayload: jws.RawPayload,
-		Payload:    jws.Payload,
-		Signature:  jws.Signature}
-
-	// Check JWT issuer details
-	issuer, audiences, err := GetJWTIssuer(jwt)
-	assert.Error(t, err)
-	assert.Nil(t, audiences)
-	assert.Equal(t, 0, len(audiences))
-	assert.Empty(t, issuer)
-}
-
-func TestGetJWTIssuerWithNoClaims(t *testing.T) {
-	// Parse the mocked JWS token.
-	token := mockGoodToken()
-	jws, err := jose.ParseJWS(token)
-	assert.NotNil(t, jws)
-	assert.NoError(t, err)
-
-	// Verify the header, payload, and signature.
-	parts := strings.Split(token, ".")
-	assert.NotNil(t, parts)
-	assert.Equal(t, parts[0], jws.RawHeader)
-	assert.Equal(t, parts[1], jws.RawPayload)
-
-	assert.NotNil(t, jws.Header)
-	assert.NotNil(t, jws.Payload)
-	assert.NotNil(t, jws.Signature)
-
-	jwt := jose.JWT{
-		RawHeader: jws.RawHeader,
-		Header:    jws.Header,
-		Signature: jws.Signature}
-
-	// Check JWT issuer details
-	issuer, audiences, err := GetJWTIssuer(jwt)
-	assert.Error(t, err)
-	assert.Nil(t, audiences)
-	assert.Equal(t, 0, len(audiences))
-	assert.Empty(t, issuer)
-}
+func TestGetJWTIssuerIdentityExpiry(t *testing.T) {
+	tests := map[int]struct {
```

Comment:
If we're using contiguous keys for indexing the map, it feels like this should be a slice instead?

```suggestion
	tests := []struct{
```

Reply:
Yeah, slice would be the right choice here; this has been fixed.